### ISSUE TITLE
Enable network CAN flag (SocketCAN)

### ISSUE THREAD

**Issue by duraki** [linuxkit/linuxkit#3172]
As pointed by docker/for-mac#2580, `linuxkit` should enable CAN flag in kernel configuration to allow Docker containers to setup virtual CAN device. The namespace was presented by O. Hartkopp in [Linux Kernel 4.12](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/log/net/can?h=linux-4.12.y); update docs available on [SocketCAN April 2018](https://drive.google.com/viewerng/viewer?url=https://wiki.automotivelinux.org/_media/agl-distro/agl2018-socketcan.pdf).

This **issue affects** both Linux & MacOS. 

With `linux-kernel@4.12` update, we can now bridge cap for virtual CAN interface in Docker, instead of relying on hardware devices and tunnelling.

https://github.com/linuxkit/linuxkit/blob/262e6ed3ec9d1d6327df58eda02c8f20f3571b93/kernel/config-4.18.x-x86_64#L1451

This wouldn't be such a hassle, if D4M (Docker 4 Mac) wasn't private. We can't recompile `docker-for-mac.iso` properly as seen in linked issue. This needs to be set on [linuxkit/linuxkit](/linuxkit/linuxkit) side.

**Comment by duraki** [comment]
Since it's my first time contributing to [linuxkit/linuxkit](/linuxkit/linuxkit), I'm allowed to ask a few questions:

* Shall we bring this topic to KSPP newsletter?
* Shall we flag on `check-kernel-config.sh` to verify?
* Shall it [officially] support landlock?

**Comment by rn** [comment]
Can't you compile the require kernel modules externally and modprobe them? All the config options in `net/can` seem to be `tristate` suggesting this should be possible

**Comment by duraki** [comment]
No. Tested under RedHat Linux and MacOS. Check [this comment](https://github.com/docker/for-mac/issues/2580#issuecomment-375726422) on why it's not possible. We need explicit allowance from linuxkit side.

**Comment by brandonros** [comment]
Can this be closed?

**Comment by duraki** [comment]
@brandonros not yet, give me some more time so I can coop with @hartkopp and make a final decision. Thank you. @rn: will provide additional information.

**Comment by thomasantony** [comment]
@duraki Were you able to get this to work with D4M ?

**Comment by PhilippKirchnerOSMA** [comment]
Are there any news on support for SocketCan in D4M?
